src/mesh/checkpoint_io.C

```diff
@@ -148,11 +150,32 @@ void CheckpointIO::write (const std::string & name)
   // do a gather_to_zero() and support that case too.
   _parallel = _parallel || !mesh.is_serial();
 
+  std::string dir_name;
+  std::string header_file_name = name;
+  std::string str_my_n_procs = "1";
+
+  if (_parallel)
+    str_my_n_procs = std::to_string(_my_n_processors);
+
+  auto ext = extension(name);
+  dir_name = name + "/" + str_my_n_procs;
+  header_file_name = dir_name + "/header" + ext;
+
+  auto errcode = Utility::mkdir(name.c_str());
+  if (errcode != 0 && errcode != -1)
+    libmesh_error_msg("Failed to create mesh split directory '" << name << "'");
+  errcode = Utility::mkdir(dir_name.c_str());
+  if (errcode == -1)
+    libMesh::out << "In CheckpointIO::write, directory "
```

Comment:
Consider using the `libmesh_warning()` macro for this, so that users who configure with `--disable-warnings` don't have to see it.